### PR TITLE
SEMI-1185  Fix error introduced with tight loop on reconnect.  This w…

### PR DESCRIPTION
**Pull request by mike-clover-com**
Fix error introduced with tight loop on reconnect.  This was mostly manifested on the cloud configuration

@clover/clover-semi-int

### CODE REVIEW

File: src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts [hunk 1]
```diff
@@ -25,6 +25,24 @@ export abstract class WebSocketCloverTransport extends CloverTransport implement
...
+     */
+	private reconnecting: boolean = false;
+
+	/**
+	 * Subclasses need to set this at times.
+	 *
+	 * @param newValue
+	 * @returns {boolean}
+     */
+	protected setReconnecting(newValue: boolean): boolean {
```
Comment (cloverStevieRay) [inline src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts]: Why is the return value the old value?

File: src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts [hunk 2]
```diff
@@ -25,6 +25,24 @@ export abstract class WebSocketCloverTransport extends CloverTransport implement
...
+		let oldValue: boolean = this.reconnecting;
+		this.reconnecting = newValue;
+		return oldValue;
```
Comment (david-clover-com) [inline src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts]: Does this need to return something?  Doesn't appear that the callers ever use the return value.

**Review by david-clover-com — APPROVED** [review]

File: src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts (hunk 1 above)
Comment (david-clover-com) [reply src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts]: Doesn't appear the callers ever use the return ...

File: src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts [hunk 3]
```diff
@@ -62,11 +80,15 @@ export abstract class WebSocketCloverTransport extends CloverTransport implement
...
-        setTimeout(this.reconnector, this.reconnectDelay);
+		// If we are already reconnecting, do not start another.
+		if(!this.reconnecting) {
+			this.setReconnecting(true);
+			if (this.shutdown) {
+				this.logger.debug("Not attempting to reconnect, shutdown...");
+				return;
+			}
+			setTimeout(this.reconnector, this.reconnectDelay);
+		}
```
Comment (cloverStevieRay) [inline src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts]: Do we need a debug log to show some of the reasons calling reconnect will not do anything.... if shutdown is set, or if already reconnecting?

File: src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts [hunk 4]
```diff
@@ -126,7 +153,7 @@ export abstract class WebSocketCloverTransport extends CloverTransport implement
 			// being dropped every 65536.
 			this.messageSenderId = setTimeout(function () {
 				this.sendMessageThread()
-			}.bind(this), 0);
+			}.bind(this), 5);
 		}
```
Comment (cloverStevieRay) [inline src/com/clover/remote/client/transport/websocket/WebSocketCloverTransport.ts]: by changing the 0 to 5 do we need to change the comment about setTimeout?

File: src/com/clover/remote/client/transport/websocket/WebSocketCloudCloverTransport.ts [hunk 5]
```diff
@@ -105,6 +105,8 @@ export class WebSocketCloudCloverTransport extends WebSocketCloverTransport {
             function(data) { this.deviceNotificationSent(data);}.bind(this),
             function(error) {
                 this.connectionError(this.webSocket, "Error sending alert to device." + error);
+                // This may end a reconnect attempt
```
Comment (cloverStevieRay) [inline src/com/clover/remote/client/transport/websocket/WebSocketCloudCloverTransport.ts:111]: maybe add a comment at the top of the initialize() method discussing the setting of reconnect to false in an error condition.

Comment (mike-clover-com) [reply src/com/clover/remote/client/transport/websocket/WebSocketCloudCloverTransport.ts:111]: Yep, did so

**Pull request merged**